src/utils/event-manager.js

```diff
@@ -0,0 +1,226 @@
+  pinch: 'pinch',
+  pinchin: 'pinch',
+  pinchout: 'pinch',
+  pinchstart: 'pinch',
+  pinchmove: 'pinch',
+  pinchend: 'pinch',
+  pinchcancel: 'pinch',
+  rotate: 'rotate',
+  rotatestart: 'rotate',
+  rotatemove: 'rotate',
+  rotateend: 'rotate',
+  rotatecancel: 'rotate',
+  pan: 'pan',
+  panstart: 'pan',
+  panmove: 'pan',
+  panup: 'pan',
+  pandown: 'pan',
+  panleft: 'pan',
+  panright: 'pan',
+  panend: 'pan',
+  pancancel: 'pan',
+  swipe: 'swipe',
+  swipeleft: 'swipe',
+  swiperight: 'swipe',
+  swipeup: 'swipe',
+  swipedown: 'swipe'
+};
+
+const RECOGNIZERS = {
+  doubletap: new Tap({
+    event: 'doubletap',
+    taps: 2
+  }),
+  pan: new Pan({
+    threshold: 10
+  }),
+  pinch: new Pinch(),
+  press: new Press(),
+  rotate: new Rotate(),
+  swipe: new Swipe(),
+  tap: new Tap()
+};
+
+/**
+ * Map gestural events typically provided by browsers
+ * that are not reported in 'hammer.input' events
+ * to corresponding Hammer.js gestures.
+ */
+const GESTURE_EVENT_ALIASES = {
+  tap: 'click'
+};
+
+/**
+ * Single API for subscribing to events about both
+ * basic input events (e.g. 'mousemove', 'touchstart', 'wheel')
+ * and gestural input (e.g. 'click', 'tap', 'panstart').
+ * Delegates event registration and handling to Hammer.js.
+ */
+class EventManager {
+  constructor(element, options) {
+    // TODO: support overriding default RECOGNIZERS by passing
+    // recognizers / configs, keyed to event name.
+
+    // how to get inputClass from createInputInstance without a Manager instance?
+    // mostly just need to run logic from createInputInstance...
+    // Issue filed: https://github.com/hammerjs/hammer.js/issues/1106
+    const inputClass = PointerMoveEventInput;
+
+    this._onBasicInput = this._onBasicInput.bind(this);
+    this.manager = new Manager(element, {
+      inputClass
+    })
+    .on('hammer.input', this._onBasicInput);
+
+    this.gestureAliases = {};
+
+    // Handle mouse wheel events as well
+    this._onWheelEvent = this._onWheelEvent.bind(this);
+    this.wheelInput = new WheelInput(element, this._onWheelEvent);
+  }
+
+  /**
+   * Register an event handler function to be called on `event`.
+   * @param {string|Object} event   An event name (String) or map of event names to handlers
+   * @param {Function} [handler]    The function to be called on `event`.
+   */
+  on(event, handler) {
+    if (typeof event === 'string') {
+      // Special handling for gestural events.
+      const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
+      if (recognizerEvent) {
+        // Add recognizer for this event if not already added.
+        if (!this.manager.get(recognizerEvent)) {
+          this.manager.add(RECOGNIZERS[recognizerEvent]);
+        }
+
+        // Alias to a recognized gesture as necessary.
+        const gestureEventAlias = GESTURE_EVENT_ALIASES[recognizerEvent];
+        if (gestureEventAlias) {
+          this.gestureAliases[event] = this._wrapAliasedGestureHandler(gestureEventAlias, handler);
```

Comment:
Should check if a wrapper is already created. Otherwise the same event will be emitted multiple times.

Also arguments supplied to `_wrapAliasedGestureHandler` does not match the method (should be `event`.